packages/now-static-build/src/utils/gatsby.ts

```diff
@@ -0,0 +1,78 @@
+import { PackageJson } from '@vercel/build-utils';
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import {
+  fileExists,
+  readPackageJson,
+  DeepWriteable,
+  writePackageJson,
+} from './_shared';
+
+const defaultConfig = {
+  plugins: [
+    {
+      resolve: 'gatsby-plugin-vercel',
+      options: {},
+    },
+  ],
+};
+
+export async function injectVercelAnalyticsPlugin(dir: string) {
+  // Gatsby requires a special variable name for environment variables to be
+  // exposed to the client-side JavaScript bundles:
+  process.env.GATSBY_VERCEL_ANALYTICS_ID = process.env.VERCEL_ANALYTICS_ID;
+
+  const gatsbyConfigName = 'gatsby-config.js';
+  const gatsbyPluginPackageName = 'gatsby-plugin-vercel';
+
+  const gatsbyConfigPath = path.join(dir, gatsbyConfigName);
+
+  const pkgJson: DeepWriteable<PackageJson> = (await readPackageJson(
+    dir
+  )) as DeepWriteable<PackageJson>;
+  if (!pkgJson.dependencies) {
+    pkgJson.dependencies = {};
+  }
+  if (!pkgJson.dependencies[gatsbyPluginPackageName]) {
+    pkgJson.dependencies[gatsbyPluginPackageName] = 'latest';
+
+    await writePackageJson(dir, pkgJson);
```

Comment:
Is this going to cause a problem with yarn.lock or package-lock.json?

Reply:
No, we don't run the command with `--frozen-lockfile` or anything by default. It'll re-generate on install.

Reply:
Also, this is feature gated strictly for Vercel Analytics users so we won't be mutating every install on our platform.

Reply:
> No, we don't run the command with --frozen-lockfile or anything by default. It'll re-generate on install.

@Timer We are going to start allowing the user to define custom install commands soon so this could happen in the future.

Reply:
When we do that we might need to actually execute `yarn add` or `npm i` then!